stego_wave/src/formats/wav.rs

```diff
@@ -0,0 +1,589 @@
+use crate::error::StegoError;
+use crate::object::{
+    AudioFileSpec, AudioSteganography, ByteIterator, ResultStego, UniqueRandomIndices,
+};
+use derive_builder::Builder;
+use std::io::Cursor;
+use std::iter;
+use std::path::{Path, PathBuf};
+
+const LSB_DEEP_DEFAULT: u8 = 1;
+const MAX_OCCUPANCY: usize = 70;
+const HEADER: &str = "STEG";
+
+const fn lsb_deep_default() -> u8 {
+    LSB_DEEP_DEFAULT
+}
+
+/// A steganography encoder/decoder for 16-bit WAV audio files.
+///
+/// This struct provides methods for hiding and extracting messages in 16-bit WAV files.
+///
+/// # Examples
+///
+/// ```rust
+/// # use stego_wave::{formats::wav::WAV16, AudioSteganography};
+///  // Create a default WAV16 instance (lsb_deep = 1)
+/// let wav16 = WAV16::default();
+///  // Create WAV16 instance with custom lsb_dep
+/// let wav16 =  WAV16::builder().lsb_deep(4).build().unwrap();
+/// ```
+#[derive(Builder, Debug, PartialEq)]
+#[builder(build_fn(validate = "Self::validate"))]
+#[builder(name = "WAV16Builder")]
+pub struct WAV16 {
+    #[builder(default = "lsb_deep_default()")]
+    lsb_deep: u8,
+}
+
+impl Default for WAV16 {
+    fn default() -> Self {
+        WAV16::builder().build().unwrap()
+    }
+}
+
+impl WAV16Builder {
+    fn validate(&self) -> Result<(), String> {
```

Comment:
According to the docs
```
Pre-build validation: You can use #[builder(build_fn(validate = "path::to::fn"))] to add your own validation before the target struct is generated.
```

Reply:
Maybe it should not be in a struct impl, but rather a separate function. This way, it looks like validation is optional